multires-lyrics-search/README.md

```diff
@@ -1,109 +1,176 @@
+
+# 🐍 Build the app with Python
+
+These instructions explain how to build the example yourself and deploy it with Python. If you want to skip the building steps and just run the app, check out the Docker section below.  
+
+## Pre requirements:
+1. You have a working Python 3.7 or 3.8 environment. 
+2. We recommend creating a [new python virtual envoriment](https://docs.python.org/3/tutorial/venv.html) to have a clean install of Jina and prevent dependency clashing.   
+3. You have at least 2GB of free space on your hard drive. 
+
+
+## Step 1. Clone the repo and install Jina
+
+Begin by cloning the repo so you can get the required files and datasets. 
+
+```sh
+git clone https://github.com/jina-ai/examples
+cd examples/multires-lyrics-search
+```
+
+On your terminal,  you should now be located in the multires-lyrics-search folder. Let's install Jina and the other required Python libraries. For futher information on installing Jina check out our [documentation](https://docs.jina.ai/chapters/core/setup/). 
+
+```sh
+pip install -r requirements.txt
+```
+If this command runs without any error messages, you can then move onto step two. 
 
-## Use toy data
+## Step 2. Download your data to search 
 
-We have included 1000 lyrics as toy data in [`toy-data`](toy-data).
-This data is ready to use with this example.
+You have two different options here. You can either use the toy-data we provide in this repo, which is quick to index but will give very poor results. Alternatively you can download a larger dataset, which takes longer to index, but you will see better results.  
 
-## [Optional] Download full lyrics dataset
+### Toy dataset
+Skip to step 3. No action needed here. 
 
-**NOTE**: This is what is used in the Docker image and is **required** if you want to build it (the Docker image) yourself. 
+### Full dataset 
+Begin by installing the Kaggle library if you haven't already done so.  You will also need to setup your API keys as [explained here](https://github.com/Kaggle/kaggle-api#api-credentials) .
+```sh
+pip install kaggle
+```
 
-If you want to use the full dataset, you can download it from kaggle (https://www.kaggle.com/neisse/scrapped-lyrics-from-6-genres).
-To get it, once you have your Kaggle Token in your system as described in (https://www.kaggle.com/docs/api), run:
+Running the following bash script should perform all the steps needed to download the full dataset. 
 
 ```bash
 bash get_data.sh
 ```
 
-## Install
+## Step 3. Index your data
 
-```bash
-pip install -r requirements.txt
+In this step, we will index our data. When we run the following command, we call our index Flow and pass our data through it. The Flow begins by breaking each song into sentences and then encoding those sentences using a language model. The data is then stored in an Index. 
```

Comment:
Question: Do we really need to write "In this step, we will index our data."? The section is already "Index your data" after all.

I can see the rationale if more background is needed, but in some cases it just re-states the section title